pkg/addons/data/ClusterRole.rbac.authorization.k8s.io/dedicated-cluster-admin.yaml

```diff
@@ -0,0 +1,167 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: dedicated-cluster-admin
+rules:
+- apiGroups:
+  - ''
+  - user.openshift.io
+  resources:
+  - groups
+  - users
+  - identities
+  - useridentitymappings
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - ''
+  - authorization.openshift.io
+  resources:
+  - rolebindings
+  - clusterrolebindings
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - rolebindings
+  - clusterrolebindings
```

Comment:
I am not sure we want to give users the power to crud clusterrolebindings.

Reply:
I posed this very question to @abhgupta, this was his response:
```
This is required to allow dedicated admins to assign/remove cluster level roles (good example would be view and self-provisioner roles) to other users. Also, we had noticed early on that the ability to get/list these resources was required to enable other functionality related to user and group management (I think - its been a while... or perhaps to run commands like who-can).
```

Reply:
@jim-minter ptal, this gives root access to end users in the cluster, I still don't think we should be doing this.

Reply:
I attempted to do this exact thing since I thought the same issue:
```
[cloud-user@master-000000 ~]$  oc login -u dedicated-cluster-admin -p <pw> --server=https://localhost:443
Login successful.
You have access to the following projects and can switch between them with 'oc project <projectname>':

    dedicated-admin
    dedicated-readers
  * test
[cloud-user@master-000000 ~]$ oc adm policy add-cluster-role-to-user cluster-admin dedicated-cluster-admin
Error from server (Forbidden): clusterrolebindings.authorization.openshift.io "cluster-admin-0" is forbidden: attempt to grant extra privileges:  ...
[cloud-user@master-000000 ~]$ oc whoami
dedicated-cluster-admin

```

Reply:
Added a test case covering escalation. https://github.com/openshift/openshift-azure/pull/570/files#diff-cd98a29fcf0b4ec8276ee320d492d41fR71